### PR TITLE
CI: Add universal build support for macOS

### PR DESCRIPTION
**Pull request by Developer-Ecosystem-Engineering**
### Description
- Update all dependencies to universal
- Enable NEON for several projects
- Support building on macOS 10.15 or macOS 11, x86, Rosetta or natively

### Motivation and Context
- OBS does not currently build universal but the dependencies would support it

### How Has This Been Tested?
Tested on macOS 11.3 M1 & Rosetta
Tested on macOS 10.15.7 x86
Xcode 12.4

### Types of changes
- New feature (non-breaking change which adds functionality - Apple silicon support)
- Performance enhancement (non-breaking change which improves efficiency - NEON support in some libraries)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Review by RytoEX — CHANGES_REQUESTED** [review]
You don't need to modify `build-script-macos-01.sh` or `build-script-macos-02.sh`.  Those files are generated from `.github/workflows/create_scripts.yml` after each commit to the master branch.

File: .github/workflows/build_deps.yml
Comment (Developer-Ecosystem-Engineering) [reply .github/workflows/build_deps.yml]: Okay thanks done!

**Comment by Developer-Ecosystem-Engineering** [comment]
> You don't need to modify `build-script-macos-01.sh` or `build-script-macos-02.sh`. Those files are generated from `.github/workflows/create_scripts.yml` after each commit to the master branch.

Was mostly to demonstrate what it generates for observers that are interested in building on their own without needing to generate it themselves. I can remove them if desired, as you point out they are auto generated anyway from the .yml (which is how they were generated for this PR of course)

**Comment by PatTheMav** [comment]
Yeah it's fine for the PR, especially as it will only be overwritten after merging iirc.

**Comment by Developer-Ecosystem-Engineering** [comment]
Chasing environment failures in the build env, will see if we can setup a local environment to replicate before subsequent amendments.

**Comment by Developer-Ecosystem-Engineering** [comment]
These changes have been integrated into https://github.com/obsproject/obs-deps/pull/60. Do we want to leave this open or just use that one?

**Comment by RytoEX** [comment]
> 
> 
> These changes have been integrated into #60. Do we want to leave this open or just use that one?

My personal preference is that PRs are narrowly scoped and reviewed on their own merits, but I'll defer to @DDRBoxman and @PatTheMav on this since it's macOS heavy.

If this is to be merged, it will need the commits appropriately squashed and the build failure resolved or explained.

**Comment by DDRBoxman** [comment]
Closing this since #60 is merged :)

**Pull request closed without merging**
